lib/model/query/entities.js

```diff
@@ -110,18 +118,20 @@ const _processSubmissionDef = (submissionDefId, submissionId) => async ({ Datase
   // If no data was returned, e.g. if create flag was not set, then don't continue
   if (!partial)
     return null;
-  return Entities.createNew(partial, submissionDef);
+  const sourceId = await Entities.createSource(submissionDefId, eventId, { submission: { instanceId: submissionDef.instanceId } });
```

Comment:
Even though we are saving instanceId in source table, we are still fetching submission.create event. What is the benefit of persisting this additional information?